pkg/cmd/label/clone.go

```diff
@@ -0,0 +1,141 @@
+package label
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/cli/cli/v2/utils"
+	"github.com/spf13/cobra"
+)
+
+type cloneOptions struct {
+	BaseRepo   func() (ghrepo.Interface, error)
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+
+	SourceRepo ghrepo.Interface
+	Force      bool
+}
+
+func newCmdClone(f *cmdutil.Factory, runF func(*cloneOptions) error) *cobra.Command {
+	opts := cloneOptions{
+		HttpClient: f.HttpClient,
+		IO:         f.IOStreams,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "clone <source [HOST/]OWNER/REPO>",
+		Short: "Clones labels from another repo",
+		Long: heredoc.Doc(`
+			Clones labels from another repo into the current repo on GitHub.
+
+			All labels from the source repo will be copied to the current repo.
+			Any labels in the current repo that are not in the source repo will
+			not be deleted or modified.
+
+			If any labels from the source repo already exist in the current repo
+			the command will fail. You can overwrite existing labels in the
+			current repo using the --force flag.
+
+		`),
+		Example: heredoc.Doc(`
+			# clone and overwrite labels from the cli/cli repo into the current repo
+			$ gh label clone cli/cli --force
+		`),
+		Args: cmdutil.ExactArgs(1, "cannot clone labels: source repo argument required"),
+		RunE: func(c *cobra.Command, args []string) error {
+			var err error
+			opts.BaseRepo = f.BaseRepo
+			opts.SourceRepo, err = ghrepo.FromFullName(args[0])
+			if err != nil {
+				return err
+			}
+			if runF != nil {
+				return runF(&opts)
+			}
+			return cloneRun(&opts)
+		},
+	}
+
+	cmd.Flags().BoolVarP(&opts.Force, "force", "f", false, "Overwrite labels in the current repo from the source repo.")
+
+	return cmd
+}
+
+func cloneRun(opts *cloneOptions) error {
+	baseRepo, err := opts.BaseRepo()
+	if err != nil {
+		return err
+	}
+
+	successCount, totalCount, err := cloneLabels(opts)
```

Comment:
In non-tty mode, for scripts, what do you think the behavior should be when not all of the labels are successfully cloned? Should that result in an error exit code or a successful one?

Reply:
Good question. Since `--force` is defined to specifically overwrite and cloning really is more of a "merging" without it, I'm inclined to *not* exit with an error code unless we failed completely...

...though, now that I opine on that, perhaps we shouldn't fail ever (even with the TTY messages) if the only error was that a duplicate name was already defined.

Thoughts?

Reply:
Yeah I agree, I think that behavior makes sense. Only exit with error status if we receive an error that is not "label already exists".